jbpm-flow/src/main/java/org/jbpm/workflow/instance/impl/WorkflowProcessInstanceImpl.java

```diff
@@ -256,13 +256,17 @@ public NodeInstance getFirstNodeInstance(final long nodeId) {
 		return result;
 	}
 
-	@Override
-	public NodeInstance getNodeInstance(final Node node) {
+    @Override
+    public NodeInstance getNodeInstance(final Node node) {
+        return getNodeInstance(node, true);
+    }
+
+    public NodeInstance getNodeInstance(final Node node, boolean unwrap) {
 	    Node actualNode = node;
 	    // async continuation handling
-	    if (node instanceof AsyncEventNode) {
+        if (node instanceof AsyncEventNode) {
             actualNode = ((AsyncEventNode) node).getActualNode();
-        } else if (useAsync(node)) {
+        } else if (unwrap && useAsync(node)) {
```

Comment:
@elguardian just a question - this should be called wrap then, shouldn't it? We are passing there false when we want it unwrapped.

Reply:
done